pkg/webui/components/map/index.js

```diff
@@ -0,0 +1,68 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import L from 'leaflet'
+
+import style from './map.styl'
+
+// Fix broken links to leaflet images.
+delete L.Icon.Default.prototype._getIconUrl
+L.Icon.Default.mergeOptions({
+  iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
+  iconUrl: require('leaflet/dist/images/marker-icon.png'),
+  shadowUrl: require('leaflet/dist/images/marker-shadow.png'),
+})
+class Map extends React.Component {
+
+  getMapCenter (markers) {
+    // This will calculate and zoom and map center long/lang based on all markers provided.
+    // Currenlty its just returns the first marker.
+    return markers[0]
+  }
+
+  componentDidMount () {
+    const { markers } = this.props
+    const { position } = ( markers.length >= 1 ) ? this.getMapCenter(markers) : markers[0]
+
+    // create map
+    this.map = L.map('map', {
+      center: [ position.latitude, position.longitude ],
+      zoom: 11,
+      layers: [
+        L.tileLayer('http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png', {
+          attribution: '&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors',
+        }),
+      ],
+      zoomControl: false,
+    })
+
+    markers.map(maker =>
+      L.marker([ maker.position.latitude, maker.position.longitude ])
+        .bindPopup( maker.name )
+        .addTo(this.map)
+    )
+  }
+
+  render () {
+    return (
+      <div className={style.mapContainer}>
+        <div className={style.map} id="map" />
+      </div>
+    )
+  }
+}
+
+export default Map
+
```

Comment:
redundant new line

Reply:
@bafonins Do you mean line 68 or added the export at the bottom of the file?

Reply:
Line 68 is redundant, we always use one final newline at the end of each file. Try setup `.editorconfig` which will take care of these things automatically.

Reply:
.editorconfig is working which is why its there, soo is this comment to add or to remove this redundant line?

Reply:
Indeed seems to be resolved now. Had an older commit before, which had two new lines at the end.